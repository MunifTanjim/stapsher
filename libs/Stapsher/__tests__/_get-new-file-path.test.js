const helpers = require('../../../__tests__/helpers')

helpers.addSnapshotSerializers()

helpers.mockDate()
helpers.mockUUIDv1()

const utils = require('../utils')
utils.getNewFilePath = jest.fn(() => 'new/file/path')

const Stapsher = require('../../Stapsher')

let mockConfig = {}

describe('libs/Stapsher:__getNewFilePath', () => {
  const parameters = helpers.getParameters()

  let stapsher
  beforeEach(() => {
    mockConfig = helpers.readConfigFile()[parameters.entryType]

    stapsher = new Stapsher(parameters)

    stapsher.config = {
      get: key => mockConfig[key]
    }
  })

  it('calls necessary functions', () => {
    stapsher.__getNewFilePath()

    expect(utils.getNewFilePath).toHaveBeenCalledTimes(1)
    expect(utils.getNewFilePath.mock.calls[0]).toMatchSnapshot()
  })

  it('returns file path', () => {
    expect(stapsher.__getNewFilePath()).toEqual(utils.getNewFilePath())
  })
})
