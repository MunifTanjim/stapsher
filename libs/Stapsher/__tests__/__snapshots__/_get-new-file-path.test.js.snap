// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`libs/Stapsher:__getNewFilePath calls necessary functions 1`] = `
[
  "data/comments",
  "0.7w0.z3r0.7w0.53v3n",
  null,
  "yaml"
]
`;
